eng/common/TestResources/New-TestResources.ps1

```diff
@@ -224,13 +227,19 @@ function SetDeploymentOutputs([string]$serviceName, [object]$azContext, [object]
                 if (ShouldMarkValueAsSecret $serviceDirectoryPrefix $key $value $notSecretValues) {
                     # Treat all ARM template output variables as secrets since "SecureString" variables do not set values.
                     # In order to mask secrets but set environment variables for any given ARM template, we set variables twice as shown below.
-                    Write-Host "##vso[task.setvariable variable=_$key;issecret=true;]$value"
-                    Write-Host "Setting variable as secret '$key': $value"
+                    if ($DevopsLogging) {
+                        Write-Host "##vso[task.setvariable variable=_$key;issecret=true;]$value"
+                        Write-Host "Setting variable as secret '$key': $value"
+                    } else {
+                        Write-Host "Setting variable as secret '$key': REDACTED"
```

Comment:
This breaks the manual use case. This script isn't just for DevOps and must be manually runnable. It's only in .NET that we do anything with these variables. I don't set the env vars directly in these scripts because:

1. No consistent way across all platforms to set persistent variables. Even in linux it depends on what shell is current or shells available (there's some roundabout ways, but it complicates the script).
2. Can't set them in the process because that would only work if the user was using PowerShelll as their current shell and simply invoked these scripts within that shell. If they ran `pwsh New-TestResources.ps1` in linux or on Windows using cmd, for example (or just ran the .ps1 file in Windows, which will create a new pwsh or powershell process) the env vars wouldn't be retained.

Reply:
All this code is nested under a condition for `$CI == $true` so it won't affect the manual case.